src/storage/heap_file.c

```diff
@@ -15832,6 +15832,34 @@ heap_rv_redo_insert (THREAD_ENTRY * thread_p, const LOG_RCV * rcv)
       return er_errid ();
     }
 
+  if (is_passive_transaction_server ())
+    {
+      if (recdes.type != REC_ASSIGN_ADDRESS)
+	{
+	  /* Classname cache (locator_Mht_classnames) status should be reverted when
+	   * DROP TABLE statement is rolled back (RVHF_COMPENSATE_WITH_INSERT).
```

Comment:
heap_rv_redo_insert function is also used in replication(redo) of RVHF_INSERT and RVHF_INSERT_NEWHOME. I think they don't need to update the status of classname entry.

Reply:
I added a condition to update the cache only for operations related to records of the root class. Operations on the root class only occur when there is a CREATE TABLE or when a DROP TABLE is rolled back

Reply:
Do you mean that 'if (OID_EQ (&classoid, oid_Root_class_oid))' can filter out all the cases of RVHF_INSERT and RVHF_INSERT_NEWHOME?

Reply:
While I initially thought there would be no REC_NEWHOME record insertion for root classes, as they are not MVCC tables, it seems that we should handle REC_NEWHOME as an exception. This is because the record size can change due to ALTER statements.

There are likely two scenarios where new records might be inserted into root classes:
1. REC_ASSIGN_ADDRESS: This appears when CREATE TABLE statements.
2. REC_NEWHOME: This appears when the record size changes during ALTER TABLE.

For other record types, such as REC_HOME, REC_BIGONE, REC_RELOCATION, we can see those types during the execution of the UNDO operation for DROP statements. It's not certain whether a root class's record can become REC_BIGONE.